src/interface/TWAnyAddress.cpp

```diff
@@ -90,6 +90,14 @@ struct TWAnyAddress* TWAnyAddressCreateWithPublicKeyFilecoinAddressType(struct T
     return new TWAnyAddress{TW::AnyAddress::createAddress(publicKey->impl, TWCoinTypeFilecoin, TWDerivationDefault, prefix)};
 }
 
+struct TWAnyAddress* TWAnyAddressCreateWithPublicKeyFiroAddressType(struct TWPublicKey* _Nonnull publicKey, enum TWFiroAddressType firoAddressType) {
+    TW::PrefixVariant prefix = std::monostate();
+    if (firoAddressType == TWFiroAddressTypeExchange) {
+        prefix = TW::ExchangePrefix();
```

Comment:
In addition to the comment above, this function could take p2pkh and(?) p2sh prefixes

Reply:
Actually, it is a kind of p2pkh address, it is only different between mainnet and testnet. If want to support testnet, we can define a testnet address type and make some change in `ExchangeAddress` for it.